File: practica6/Item.cpp
#include "Item.h"
#include <iostream>
#include "global.h"

Item::Item(string n, int e, int u) : nombre(n), efectoCuracion(e), usos(u) {}

string Item::getNombre() const { return nombre; }
int Item::getEfecto() const { return efectoCuracion; }
int Item::getUsos() const { return usos; }
bool Item::disponible() const { return usos > 0; }

void Item::usar(Pokemon& p) {
    if (usos > 0) {
        cout << GREEN << "Se usa " << nombre << " en " 
             << p.getNombre() << " +" << efectoCuracion 
             << " de vida!" << RESET << endl;
        p.curar(efectoCuracion);
        usos--;
    } else {
        cout << RED << "No quedan " << nombre << "!" << RESET << endl;
    }
}

void Item::mostrar() const {
    cout << nombre << " (cura " << efectoCuracion 
         << ", usos: " << usos << ")";
}

File: pokemon/Item.java
package pokemon;

import static pokemon.Global.*;

/**
 * Clase que representa un item consumible en el juego
 */
public class Item {
    private String nombre;
    private int efectoCuracion;
    private int usos;
    
    public Item(String nombre, int efectoCuracion, int usos) {
        this.nombre = nombre;
        this.efectoCuracion = efectoCuracion;
        this.usos = usos;
    }
    
    // Getters
    public String getNombre() {
        return nombre;
    }
    
    public int getEfecto() {
        return efectoCuracion;
    }
    
    public int getUsos() {
        return usos;
    }
    
    public boolean disponible() {
        return usos > 0;
    }
    
    public void usar(Pokemon pokemon) {
        if (usos > 0) {
            System.out.println(GREEN + "Se usa " + nombre + " en " +
                             pokemon.getNombre() + " +" + efectoCuracion +
                             " de vida!" + RESET);
            pokemon.curar(efectoCuracion);
            usos--;
        } else {
            System.out.println(RED + "¡No quedan " + nombre + "!" + RESET);
        }
    }
    
    public void mostrar() {
        System.out.print(nombre + " (cura " + efectoCuracion +
                        ", usos: " + usos + ")");
    }
}